plugins/inputs/zfs/zfs_linux.go

```diff
@@ -68,13 +68,17 @@ func getPools(kstatPath string) ([]poolInfo, error) {
 }
 
 func getTags(pools []poolInfo) map[string]string {
-	var poolNames string
-
-	for _, pool := range pools {
-		if len(poolNames) != 0 {
-			poolNames += "::"
+	poolNames := ""
+	knownPools := make(map[string]bool)
```

Comment:
Could we use `map[string]struct{}` instead? With Go `struct{}` (but not the pointer type) takes no memory from what I understand.

Reply:
sure - changed after reading through [this thread](https://www.reddit.com/r/golang/comments/gnge5m/stop_using_mapstringstruct_for_existence_checks/) on some pros/cons

Reply:
Cool. I think you'll want to change the conditional to check the `ok` return value instead of the struct value.

Reply:
oops thanks again!